core/src/main/java/io/grpc/inprocess/InProcessTransport.java

```diff
@@ -123,7 +123,9 @@ public synchronized ClientStream newStream(
       return new NoopClientStream() {
         @Override
         public void start(ClientStreamListener listener) {
-          listener.closed(shutdownStatus, new Metadata());
+          synchronized (InProcessTransport.this) {
+            listener.closed(shutdownStatus, new Metadata());
```

Comment:
Nowhere else in this file are we synchronizing on `InProcessTransport` before calling `listener` (we do synchronize on the streams, though), and that doesn't seem like something we want to start doing.